### PR TITLE
Add types to style mixins

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Adds/corrects types to the style mixins.

## How has this been tested?
Typecheck run and usages confirmed.

## Types of changes
New/updated types.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. 

Blocks https://github.com/WordPress/gutenberg/pull/26366

### CODE REVIEW

File: packages/components/src/utils/hooks/use-controlled-state.js [hunk 1]
```diff
@@ -60,7 +67,10 @@ function useControlledState( currentState, options = defaultOptions ) {
 		fallback
 	);
 
-	const setState = ( nextState ) => {
+	const setState = (
+		/** @type {any} */
```
Comment (saramarcondes) [inline packages/components/src/utils/hooks/use-controlled-state.js:71]: The template `T` isn't in scope here 😕

File: packages/components/src/utils/hooks/use-jump-step.js [hunk 2]
```diff
@@ -29,7 +29,10 @@ function useJumpStep( {
 	const [ isShiftKey, setIsShiftKey ] = useState( false );
 
 	useEffect( () => {
-		const handleShiftKeyToggle = ( event ) => {
+		const handleShiftKeyToggle = (
+			/** @type {KeyboardEvent} */
```
Comment (saramarcondes) [inline packages/components/src/utils/hooks/use-jump-step.js:33]: This get's passed to `window.addEventListener` so we do _not_ use the synthetic version from `react`

File: packages/components/src/utils/math.js [hunk 3]
```diff
@@ -19,24 +19,23 @@ export function getNumber( value ) {
...
  * Safely adds 2 values.
  *
- * @param {number|string} args Values to add together.
+ * @param {(number|string)[]} args Values to add together.
  *
  * @return {number} The sum of values.
  */
 export function add( ...args ) {
-	return args.reduce( ( sum, arg ) => sum + getNumber( arg ), 0 );
+	return args.map( getNumber ).reduce( ( sum, arg ) => sum + arg, 0 );
```
Comment (saramarcondes) [inline packages/components/src/utils/math.js:27]: This and the change below should be equivalent in runtime but appeases the typechecking (I couldn't figure out how to communicate to TS through JSDoc that `reduce` was going to be called as `reduce<string|number, number>`. It insisted that the return type of `reduce` be `string|number`. Mapping the array to `number` from `string|number` first avoids this. From a performance perspective I think this is negligible (you'd only start to notice if we were adding several thousand numbers, right?)

**Comment by ItsJonQ** [comment]
This is exciting ❤️ ! However, my TS knowledge doesn't extend this far 🙈 - to be specific, typing things purely with TSDoc/JSDoc

**Comment by saramarcondes** [comment]
Closing as duplicate of #26078

**Pull request closed without merging**